subgraphs/tornado-cash/protocols/tornado-cash/config/networks/bsc/bsc.ts

```diff
@@ -0,0 +1,42 @@
+import { Configurations } from "../../../../../configurations/configurations/interface";
+import {
+  Network,
+  PROTOCOL_SCHEMA_VERSION,
+  PROTOCOL_SUBGRAPH_VERSION,
+  PROTOCOL_METHODOLOGY_VERSION,
+  PROTOCOL_NAME,
+  PROTOCOL_SLUG,
+  FACTORY_ADDRESS,
+  TORN_ADDRESS,
+} from "../../../../../src/common/constants";
+import { RewardIntervalType } from "../../../../../src/common/rewards";
+
+export class TornadoCashBscConfigurations implements Configurations {
+  getNetwork(): string {
+    return Network.BSC;
+  }
+  getSchemaVersion(): string {
+    return PROTOCOL_SCHEMA_VERSION;
+  }
+  getSubgraphVersion(): string {
+    return PROTOCOL_SUBGRAPH_VERSION;
+  }
+  getMethodologyVersion(): string {
+    return PROTOCOL_METHODOLOGY_VERSION;
+  }
+  getProtocolName(): string {
+    return PROTOCOL_NAME;
+  }
+  getProtocolSlug(): string {
+    return PROTOCOL_SLUG;
+  }
+  getFactoryAddress(): string {
+    return FACTORY_ADDRESS.get(Network.BSC)!.toHexString();
```

Comment:
Since you are already abstracting network-specific configurations behind an interface, would it make sense to remove the FACTORY_ADDRESS constants from `constants.ts`. Instead, declare them here and don't export them. If at any point the factory address is needed it can be accessed through `configurations.getFactoryAddress()`, but never accessing the constant directly.